test/system/250-systemd.bats

```diff
@@ -40,12 +46,13 @@ function service_setup() {
     # stdout + stderr, that warning goes to the unit file. (Today's systemd
     # is forgiving about that, but RHEL8 systemd chokes with EINVAL)
     (
-        cd $UNIT_DIR
+        cd $PODMAN_TMPDIR
         run_podman generate systemd --files --name \
                -e http_proxy -e https_proxy -e no_proxy \
                -e HTTP_PROXY -e HTTPS_PROXY -e NO_PROXY \
                --new $cname
-        mv "container-$cname.service" $UNIT_FILE
+        mv -Z "container-$cname.service" $UNIT_FILE.tmp.$$ && \
+            mv -Z $UNIT_FILE.tmp.$$ $UNIT_FILE
```

Comment:
I am confused, why do we move to a tmp file just to move to the real location directly after? I do not see th epoint of doing this vs one single mv command?

Reply:
Ultraparanoia. `mv` is not atomic across filesystems, so this is theoretically possible:
1. mv starts writing foo.service
2. another parallel job runs systemctl daemon-reload
3. systemd sees (incomplete) foo.service and does whatever systemd does with it and bad things happen
4. mv completes

Even with all this paranoia, #24010 still triggers (on my laptop), so I'm pretty sure that the fragment-file flake has nothing to do with corrupt systemd service files ... but I just want to eliminate every last possibility of test errors.

Reply:
Ok got it but then please comment the reason here because it is not obvious otherwise.